File: appcommons/src/main/java/ng/kingsley/android/preference/SimplePreference.java
package ng.kingsley.android.preference;

import android.app.Application;
import android.content.Context;
import android.content.SharedPreferences;

import com.google.gson.Gson;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * @author ADIO Kingsley O.
 * @since 13 Aug, 2015
 */
@Singleton
public class SimplePreference {

    private final Context mContext;
    private final Gson GSON;

    @Inject
    public SimplePreference(Application context, Gson gson) {
        mContext = context;
        GSON = gson;
    }

    public PreferenceStore with(String domain) {
        return new PreferenceStore(mContext, GSON, domain);
    }

    public PreferenceStore with(SharedPreferences wrapped) {
        return new PreferenceStore(wrapped, GSON);
    }
}

File: preference/src/main/java/com/kingsleyadio/appcommons/preference/SimplePreference.kt
package com.kingsleyadio.appcommons.preference

import android.app.Application
import android.content.Context
import android.content.SharedPreferences

/**
 * @author ADIO Kingsley O.
 * @since 13 Aug, 2015
 */
class SimplePreference(private val context: Application, private val converter: Converter) {

    fun with(domain: String): PreferenceStore {
        val preference = context.getSharedPreferences(domain, Context.MODE_PRIVATE)
        return PreferenceStore(preference, converter)
    }

    fun with(wrapped: SharedPreferences): PreferenceStore {
        return PreferenceStore(wrapped, converter)
    }

//    private fun name(context: Context, domain: String?): String {
//        var domain = domain
//        domain = domain?.trim { it <= ' ' } ?: ""
//        return DigestUtils.md5(context.packageName + domain)
//    }
}
